test/parallel/test-vm-measure-memory.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const vm = require('vm');
+const {
+  SUMMARY,
+  DETAILED
+} = vm.constants.measureMemory.mode;
+
+common.expectWarning('ExperimentalWarning',
+                     'vm.measureMemory is an experimental feature. ' +
+                     'This feature could change at any time');
+
+// Test measuring memory of the current context
+{
+  vm.measureMemory(undefined)
+    .then((result) => {
+      assert(result instanceof Object);
+    });
+
+  vm.measureMemory({})
+    .then((result) => {
+      assert(result instanceof Object);
+    });
+
+  vm.measureMemory({ mode: SUMMARY })
+    .then((result) => {
+      assert(result instanceof Object);
+    });
+
+  vm.measureMemory({ mode: DETAILED })
+    .then((result) => {
+      assert(result instanceof Object);
+    });
```

Comment:
Nit: can be simplified to (also for sandox)
```js
[undefined, {}, { mode: SUMMARY }, { mode: DETAILED }].forEach((options) => {
  vm.measureMemory(options).then((result) => {
    assert(result instanceof Object);
  });
});
```

Reply:
I prefer not to simplify tests with loops and keep them as dumb as possible in general as that would make it more difficult to debug from logs